app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala

```diff
@@ -423,26 +423,46 @@ class SubscriptionService @Inject() (
     Future.successful(Json.obj("error" -> errorMessage))
   }
 
+  def getOrEmptyString[T](option: Option[T]): String = option match {
+    case Some(value) if !value.toString.isEmpty => value.toString
+    case _                                      => "N/A"
+  }
+
+  def getNonEmptyOrNA(value: String): String =
+    if (value.nonEmpty) value else "N/A"
+
   private def extractSubscriptionData(id: String, sub: SubscriptionSuccess): Future[JsValue] = {
     val userAnswers = UserAnswers(id, Json.obj())
 
-    val registrationInfo = RegistrationInfo(
-      crn = sub.upeDetails.customerIdentification1.getOrElse(" "),
-      utr = sub.upeDetails.customerIdentification2.getOrElse(" "),
-      safeId = sub.upeDetails.safeId.getOrElse(""),
-      registrationDate = Option(sub.upeDetails.registrationDate),
-      filingMember = Option(sub.upeDetails.filingMember)
+    val dashboardInfo = DashboardInfo(
+      organisationName = sub.upeDetails.organisationName,
+      registrationDate = sub.upeDetails.registrationDate
     )
 
-    val ukAddress = UKAddress(
+    val nonUKAddress = NonUKAddress(
       addressLine1 = sub.upeCorrespAddressDetails.addressLine1,
-      addressLine2 = sub.upeCorrespAddressDetails.addressLine2,
-      addressLine3 = sub.upeCorrespAddressDetails.addressLine3.getOrElse(""),
-      addressLine4 = sub.upeCorrespAddressDetails.addressLine4,
-      postalCode = sub.upeCorrespAddressDetails.postCode.getOrElse(""),
+      addressLine2 = sub.upeCorrespAddressDetails.addressLine2.filter(_.nonEmpty).orElse(Some("N/A")),
+      addressLine3 = sub.upeCorrespAddressDetails.addressLine3
+        .collect {
+          case str if str.nonEmpty => str
+        }
+        .getOrElse("N/A"),
+      addressLine4 = sub.upeCorrespAddressDetails.addressLine4.filter(_.nonEmpty).orElse(Some("N/A")),
+      postalCode = sub.upeCorrespAddressDetails.postCode.filter(_.nonEmpty).orElse(Some("N/A")),
```

Comment:
Did we not have only AddressLine3 optional problem

Reply:
No the date for NonUKAddress which has optional fileds  addressLine2, addressLine4 and postalCode gets its data from UpeCorrespAddressDetails which has addressLine2 - addressLine4 and postCode as optional

Reply:
Got rid of the getOrElse